plugins/templates/create.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+const boom = require('boom');
+const schema = require('screwdriver-data-schema');
+const hoek = require('hoek');
+const urlLib = require('url');
+
+module.exports = () => ({
+    method: 'POST',
+    path: '/templates',
```

Comment:
Actually the more I think about it, the more I incline to make this `POST /templates`. 

**The only time a new version should be published is when something in `template.yaml` changes.** (Please correct me if I'm wrong @stjohnjohnson, I remember this from our conversation) If this is true, if they want to publish a new version, they need to change the `template.yaml` anyway. In that case, they can change the `version` (major, minor, and patch) in the yaml as well. We don't need the "smart detection" that @stjohnjohnson  mentioned about not publishing if nothing in `template.yaml` changes. 

I'm not in favor of `PUT /templates/:name/:version` because we need to detect what exact version to put. If we don't want to do the auto-bumping version for the users, then the user needs to know exactly what version it is. And "user" in this case is actually `template-template` (our template to publish templates), which means it would need to change every time, and that doesn't make sense to me. 

@FenrirUnbound  @petey  @stjohnjohnson